### ISSUE TITLE
Mixer post-install fails to complete when RBAC is disabled

### ISSUE THREAD

**Issue by ymesika** [istio/istio#6582]
When installing Istio with `global.rbacEnabled=false` istio-mixer-post-install keeps restarting with the following repetitive errors in failed logs:

```sh
Error from server (InternalError): error when creating "/tmp/mixer/custom-resources.yaml": Internal error occurred: failed calling admission webhook "mixer.validation.istio.io": Post https://istio-galley.istio-system.svc:443/admitmixer: x509: certificate is not valid for any names, but wanted to match istio-galley.istio-system.svc
```

**Comment by costinm** [comment]
See the other bug on not enabling it by default. 

For the case user enables it - we need a test suite and this bug fixed.

**Comment by sdake** [comment]
@costinm can you link the issue where it was suggested RBAC should be disabled by default?  I'm not sure what the use case is for not installing with RBAC all the time.

Cheers
-steve

**Comment by sdake** [comment]
@ayj I ran this against PR https://github.com/istio/istio/pull/6595 with RBAC disabled.  With RBAC enabled, things work fine.

I have the same result as @ymesika :
```
sdake@falkor-07:~$ kubectl logs -n istio-system istio-mixer-post-install-xmdrf
2018-06-26 03:26:28.564933 I | proto: duplicate proto type registered: google.protobuf.Any
2018-06-26 03:26:28.565090 I | proto: duplicate proto type registered: google.protobuf.Duration
2018-06-26 03:26:28.565123 I | proto: duplicate proto type registered: google.protobuf.Timestamp
kubernetes "attributes" configured
Error from server (InternalError): error when creating "/tmp/mixer/custom-resources.yaml": Internal error occurred: failed calling admission webhook "mixer.validation.istio.io": Post https://istio-galley.istio-system.svc:443/admitmixer?timeout=30s: x509: certificate is not valid for any names, but wanted to match istio-galley.istio-system.svc
Error from server (InternalError): error when creating "/tmp/mixer/custom-resources.yaml": Internal error occurred: failed calling admission webhook "mixer.validation.istio.io": Post https://istio-galley.istio-system.svc:443/admitmixer?timeout=30s: x509: certificate is not valid for any names, but wanted to match istio-galley.istio-system.svc
```

**Comment by sdake** [comment]
caBundle is "" - is this correct?
```
sdake@falkor-07:~/go/src/istio.io/istio/install$ kubectl get validatingwebhookconfigurations istio-galley -n istio-system -o yaml
apiVersion: admissionregistration.k8s.io/v1beta1
kind: ValidatingWebhookConfiguration
metadata:
  creationTimestamp: 2018-06-26T03:20:23Z
  generation: 1
  labels:
    app: istio-galley
    chart: galley-0.8.0
    heritage: Tiller
    release: istio
  name: istio-galley
  resourceVersion: "1021"
  selfLink: /apis/admissionregistration.k8s.io/v1beta1/validatingwebhookconfigurations/istio-galley
  uid: dd2c2b7d-78ef-11e8-a6e2-003a7dd0301e
webhooks:
- clientConfig:
    caBundle: ""

...

```

**Comment by sdake** [comment]
@smawson @jasmine-jaksic IMO this should be a P0.  Alternatives to actually fixing this problem include:

1. Documenting that the flag --set global.rbacEnabled=false does not work and will yield a nonfunctional Istio
1. Removal of the global.rbacEnabled flag from the templates

What is the use case for running without RBAC?  I get in Kubernetes 1.6 RBAC was painful, but in 1.9/1.10 RBAC is rock solid and should just always be on IMO (option 2).

Cheers
-steve

**Comment by ymesika** [comment]
+1 Removal
Will also cleanup all charts with those if enabled conditions.

**Comment by ayj** [comment]
> caBundle is "" - is this correct?

Yes (ish). The sidecar injector and config validation webhooks patch `caBundle` with the caCert data when running once certs are provisioned. With `rbacEnabled=false`, use of service accounts is disabled and thus Citadel provisioned certs are not generated for the webhooks. 

The [Istio Webhook Cert/Key Management]( https://goo.gl/tVJRdY) design requires the precedence of Citadel (Istio CA) and use of service-accounts as part of the means for cert provisioning and distribution. See [this](https://docs.google.com/a/google.com/document/d/1qOsp6qhllLpHsT-pSa211Lc146iGjjrHPY3NeMauXPw/edit?disco=AAAABfWHaaU) comment thread for the explicit requirement.

**Comment by smawson** [comment]
I'm OK saying RBAC is required, although I'm a bit worried about the spaghetti we're getting ourselves into here where everything depends on everything else. We need to figure out our layering / composability story. For 1.0 we may be too late to really break things apart in that way, so I'm OK removing those options (e.g. turning off RBAC).

BTW we also clearly need to do a better job of tracking what options need to be tested with each other so that we actually test the supported configurations of the system. We should probably list what those supported configurations are first :)

Does anyone feel strongly we should leave in the option of turning off rbac for 1.0?

**Comment by rshriram** [comment]
I thikn we should literally get rid of that option to avoid confusion. And then document that a kubernetes cluster needs to have RBAC enabled in order for Istio to work properly. (and this is Kubernetes RBAC and not Istio RBAC :) )

**Comment by ayj** [comment]
FWIW, sidecar-injector should have a similar problem when running with RBAC disabled.

**Comment by sdake** [comment]
I'll work on a patch - its a few hours of work at most.  If we decide to remove the option, we can just merge that patch :)

Cheers
-steve

**Comment by sdake** [comment]
@smawson the proposal on the table is to remove the flag `rbacEnabled=true` and replacing that with always enabling Kubernetes RBAC for Istio.  I can't think of a use case for `rbacEnabled=false`, unless someone has some form of degenerate Kubernetes cluster that doesn't have an rbac implementation.  With the choice for rbacEnabled, the code is a lot more complicated as there are 42 additional conditionals to support this option:

```
sdake@falkor-07:~/go/src/istio.io/istio/install/kubernetes/helm/istio$ grep -r rbacEnabled charts templates | wc -l
42
```

Cheers
-steve

**Comment by kyessenov** [comment]
It is surprising that it is so hard to disable RBAC. FWIW I prefer not to have RBAC since I test on a personal GKE cluster from GCE dev VM, and getting RBAC to work in this case is difficult (due to various complexities around GCE service account vs GKE service account).

**Comment by sdake** [comment]
https://github.com/istio/istio/pull/6657

**Comment by sdake** [comment]
@kyessenov I think its not super hard to disable RBAC - although in the particular case of this bug, I don't know how to fix the root problem.  Certainly removing the ability to disable RBAC as done in https://github.com/istio/istio/pull/6657 is one approach.

Cheers
-steve

**Comment by ymesika** [comment]
"Fixed" by #6657
